dask/dataframe/tests/test_merge_column_and_index.py

```diff
@@ -147,7 +147,6 @@ def test_merge_known_to_unknown(
     # Assertions
     assert_eq(result, expected)
     assert_eq(result.divisions, tuple(None for _ in range(11)))
-    assert len(result.__dask_graph__()) >= 400
```

Comment:
Do you know the motivation for these checks @TomAugspurger ? (or @jonmmease, who initially added them ?)

It seems that we are changing the graph size here, so the assertions are failing.

Reply:
I believe the graph size checks were added in cases where the merge algorithm should be able to avoid performing a full shuffle (it should be possible to perform the operation by just lining up the corresponding partitions).  The sizes were chosen empirically to be something smaller than the task graph that would be required to perform the same operation using a full shuffle.

Without understanding what this PR does, I'd say make sure that it's still the case that a full shuffle is not happening, and then bump up the number as needed.

Reply:
Great - Thanks for the info! I'll make sure we are still avoiding the full shuffle

Reply:
I just got a chance to look at these tests again. It seems that the assertion is making sure that the full shuffle **is** being performed (by checking that the graph is "large enough"), rather than the other way around. That is, the divisions are not known on one side, so the merge will require shuffling.  Since the changes in this PR result in a simpler shuffle for these cases, the graph is now smaller than "400" (more like 390).

@TomAugspurger - Does it make sense to change this check to `>=390` ? Or is it best to leave out the graph-size check moving forward?

Reply:
Mmm, let's just check >= 390 :)